src/gpl/src/initialPlace.cpp

```diff
@@ -106,23 +109,56 @@ void InitialPlace::doBicgstabPlace() {
 
   // set ExtId for idx reference // easy recovery
   setPlaceInstExtId();
-  for(int i=1; i<=ipVars_.maxIter; i++) {
-    updatePinInfo();
-    createSparseMatrix();
 
-    // BiCGSTAB solver for initial place
-    BiCGSTAB< SMatrix, IdentityPreconditioner > solver;
-    solver.setMaxIterations(ipVars_.maxSolverIter);
-    solver.compute(placeInstForceMatrixX_);
-    instLocVecX_ = solver.solveWithGuess(fixedInstForceVecX_, instLocVecX_);
-    errorX = solver.error();
+  for(size_t iter=1; iter<=ipVars_.maxIter; iter++) {
 
-    solver.compute(placeInstForceMatrixY_);
-    instLocVecY_ = solver.solveWithGuess(fixedInstForceVecY_, instLocVecY_);
-    errorY = solver.error();
+    updatePinInfo();
+    createSparseMatrix();
+    #if CUDA
```

Comment:
Even if you compile with CUDA the user should still have a runtime option to use the cpu instead (at a minimum for testing).